server/mem_agent/engine.py

```
@@ -332,3 +347,535 @@ def _subprocess_entry() -> None:
+            return
+
+        os.makedirs(self.venv_path, exist_ok=True)
+
+        # Create session-specific temp directory
+        os.makedirs(self.session_temp_path, exist_ok=True)
+
+        # Check if venv already exists
+        python_path = self._get_python_path()
+        if not os.path.exists(python_path):
+            # Create venv using system Python
+            subprocess.run(
```

Ruff (0.14.13)

`subprocess` call: check for execution of untrusted input

(S603)